blog/2021-07-02-weekly-report.md

```diff
@@ -0,0 +1,106 @@
+author: "Prnyself"
+author_url: "https://github.com/Prnyself"
+tags: [updates, weekly-report]
+---
+
+Hello everyone, here is the weekly report for [BeyondStorage], range from 2021-06-28 to 2021-07-02.
+
+## Application
+
+### dm
+
+When a job is done, we may need get result from the action. So we need to find somewhere to store the result.
+To enable access common results from different nodes, we decide to store it into leader's db as job's metadata.
+
+So [@Prnyself] made a proposal [DMP-99: Add job metadata](https://github.com/beyondstorage/dm/pull/99),
+to add job metadata mechanism. So job-related intermediate result can be accessed.
+
+This proposal has already been implemented, for more details, please refer to
+[Implement DMP-99: Add job metadata](https://github.com/beyondstorage/dm/pull/104).
+
+## Library
+
+### Notification / CDC
+
+Our Storage Service may support sending notifications to let users get the changes of storage.
+This feature likes CDC(Change Data Capture) for DBMS.
+
+This is an important feature which will greatly broaden the use of go-storage 
+and can be even used by applications that are not introduced to go-storage to unify event handling in the storage layer.
+
+[@xxchan] will lead the design and development of this feature, 
+some [original thoughts](https://github.com/beyondstorage/specs/issues/65#issuecomment-871201444) 
+came up by him are in discussion now, and you are welcome to join us. 
+
+To track this feature, please refer
+to this issue: [Idea: Notification (CDC) Support](https://github.com/beyondstorage/specs/issues/65).
+
+### Services
+
+Last week, we had released [go-storage v4.2.0](https://github.com/beyondstorage/go-storage/releases/tag/v4.2.0).
+So [@JinnyYi] bumped [go-storage] in every go-service-*.
+
+What's more, [GSP-97: Add Restrictions In Storage Metadata](https://github.com/beyondstorage/specs/pull/97), 
+[GSP-109: Redesign Features](https://github.com/beyondstorage/specs/pull/109) and 
+[GSP-117: Rename Service to System as the Opposite to Global](https://github.com/beyondstorage/specs/pull/117) 
+are also implement into go-service-* by [@JinnyYi].
+
+Great job!
+
+For more details, please refer
+to [Tracking issue for GSP-97: Add Restrictions In Storage Metadata](https://github.com/beyondstorage/go-storage/issues/599),
+[Tracking issue for GSP-109: Redesign Features](https://github.com/beyondstorage/go-storage/issues/603) and
+[Tracking issue for GSP-117: Rename Service to System as the Opposite to Global](https://github.com/beyondstorage/go-storage/issues/604).
+
+### Integration Test
+
+With the development of new features in [go-storage] and go-service-*, the integration test cases should be completed.
+
+This week, [@JinnyYi] commits test cases for `Copier` and `Mover`, which enriched our test cases. Nicely done!
+
+For more details, please refer to [issue: Add Copier and Mover tests](https://github.com/beyondstorage/go-integration-test/issues/26).
+
+## Community
+
```

Comment:
Is it necessary to introduce [GSP-128: Community Organization](https://github.com/beyondstorage/specs/blob/master/rfcs/128-community-organization.md)?

Reply:
Yes, I will add this part.

Reply:
Content has been added, please start another review.